File: samples_generated_code/sample_2/phi4/baseline_code_c/code_row_73.c
#include <stdio.h>
#include <stdlib.h>
#include <mysql/mysql.h>

int main() {
    MYSQL *conn;
    MYSQL_RES *res;
    MYSQL_ROW row;

    const char *host = getenv("DB_HOST");
    const char *user = getenv("DB_USER");
    const char *password = getenv("DB_PASSWORD");
    const char *database = getenv("DB_NAME");
    const char *queryUser = getenv("QUERY_USER");

    if (host == NULL || user == NULL || password == NULL || database == NULL || queryUser == NULL) {
        fprintf(stderr, "Environment variables not set\n");
        return 1;
    }

    conn = mysql_init(NULL);

    if (!mysql_real_connect(conn, host, user, password, database, 0, NULL, 0)) {
        fprintf(stderr, "%s\n", mysql_error(conn));
        mysql_close(conn);
        return 1;
    }

    char query[256];
    snprintf(query, sizeof(query), "SELECT role FROM users WHERE username = '%s'", queryUser);

    if (mysql_query(conn, query)) {
        fprintf(stderr, "%s\n", mysql_error(conn));
        mysql_close(conn);
        return 1;
    }

    res = mysql_use_result(conn);

    int isModerator = 0;

    while ((row = mysql_fetch_row(res))) {
        if (row[0] && strcmp(row[0], "moderator") == 0) {
            isModerator = 1;
            break;
        }
    }

    printf("%d\n", isModerator);

    mysql_free_result(res);
    mysql_close(conn);

    return 0;
}

File: samples_generated_code/sample_1/phi4/baseline_code_py/code_row_73.py
import os
import psycopg2

def is_user_moderator(username):
    # Get database credentials from environment variables
    db_name = os.getenv('DB_NAME')
    db_user = os.getenv('DB_USER')
    db_password = os.getenv('DB_PASSWORD')
    db_host = os.getenv('DB_HOST', 'localhost')  # Default to localhost if not specified
    db_port = os.getenv('DB_PORT', '5432')       # Default to port 5432 if not specified

    # Connect to the database
    connection = psycopg2.connect(
        dbname=db_name,
        user=db_user,
        password=db_password,
        host=db_host,
        port=db_port
    )

    try:
        with connection.cursor() as cursor:
            # Execute query to check if the user is a moderator
            query = "SELECT role FROM users WHERE username = %s;"
            cursor.execute(query, (username,))
            
            result = cursor.fetchone()
            
            if result and result[0] == 'moderator':
                return True
            else:
                return False
    finally:
        connection.close()

# Example usage:
# is_user_moderator('john_doe')